crates/sui-config/src/dynamic_transaction_signing_checks.rs

```diff
@@ -0,0 +1,273 @@
+            loaded_program: starlark_program,
+        })
+    }
+
+    /// Run the Starlark program in `self` with the given transaction data, signatures, input
+    /// object kinds, and receiving objects.
+    pub fn run_predicate(
+        &self,
+        tx_data: &TransactionData,
+        tx_signatures: &[GenericSignature],
+        input_object_kinds: &[InputObjectKind],
+        receiving_objects: &[ObjectRef],
+    ) -> Result<(), DynamicCheckRunnerError> {
+        let tx_data_json = serde_json::to_value(tx_data)
+            .map_err(|e| DynamicCheckRunnerError::JSONSerializationError(e.to_string()))?;
+        let tx_signatures_json = serde_json::to_value(tx_signatures)
+            .map_err(|e| DynamicCheckRunnerError::JSONSerializationError(e.to_string()))?;
+        let input_object_kinds_json = serde_json::to_value(input_object_kinds)
+            .map_err(|e| DynamicCheckRunnerError::JSONSerializationError(e.to_string()))?;
+        let receiving_objects_json = serde_json::to_value(receiving_objects)
+            .map_err(|e| DynamicCheckRunnerError::JSONSerializationError(e.to_string()))?;
+        self.run_starlark_predicate(
+            &tx_data_json,
+            &tx_signatures_json,
+            &input_object_kinds_json,
+            &receiving_objects_json,
+        )
+    }
+
+    fn run_starlark_predicate(
+        &self,
+        tx_data: &JsonValue,
+        tx_signatures: &JsonValue,
+        tx_input_object_kinds: &JsonValue,
+        tx_receiving_objects: &JsonValue,
+    ) -> Result<(), DynamicCheckRunnerError> {
+        let heap = Heap::new();
+        let env = Module::new();
+
+        let tx_data_value = Self::json_to_starlark(tx_data, &heap, 0)?;
+        let tx_signers_value = Self::json_to_starlark(tx_signatures, &heap, 0)?;
+        let tx_input_object_kinds_value = Self::json_to_starlark(tx_input_object_kinds, &heap, 0)?;
+        let tx_receiving_objects_value = Self::json_to_starlark(tx_receiving_objects, &heap, 0)?;
+
+        env.set(TX_DATA_NAME, tx_data_value);
+        env.set(TX_SIGNERS_NAME, tx_signers_value);
+        env.set(TX_INPUT_OBJECTS_NAME, tx_input_object_kinds_value);
+        env.set(TX_RECEIVING_OBJECTS_NAME, tx_receiving_objects_value);
+
+        let mut evaluator = Evaluator::new(&env);
+        let output_value = evaluator
+            .eval_module(self.module.clone(), &self.globals)
+            .map_err(|e| DynamicCheckRunnerError::ExecutionError(e.into_kind()))?;
+        let transaction_allowed = output_value
+            .unpack_bool()
+            .ok_or_else(|| DynamicCheckRunnerError::InvalidReturnType(output_value.to_repr()))?;
+        if transaction_allowed {
+            Ok(())
+        } else {
+            Err(DynamicCheckRunnerError::CheckFailure)
+        }
+    }
+
+    fn json_to_starlark<'v>(
+        value: &JsonValue,
+        heap: &'v Heap,
+        depth: u64,
+    ) -> Result<Value<'v>, DynamicCheckRunnerError> {
+        if depth >= MAX_DEPTH {
```

Comment:
Assuming no issues from the TypeInput question. Do we need these sort of checks? Isn't all of this structure specified by us?

Reply:
I just worry that we have put something in the way for us to trip over in the case of an emergency

Reply:
Hmm this is very true. You've convinced to remove this I think.